fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -347,25 +347,32 @@ Future<Standalone<RangeResultRef>> ConflictingKeysImpl::getRange(ReadYourWritesT
 }
 
 ACTOR Future<Standalone<RangeResultRef>> ddMetricsGetRangeActor(ReadYourWritesTransaction* ryw, KeyRangeRef kr) {
-	try {
-		auto keys = kr.removePrefix(ddStatsRange.begin);
-		Standalone<VectorRef<DDMetricsRef>> resultWithoutPrefix =
-		    wait(waitDataDistributionMetricsList(ryw->getDatabase(), keys, CLIENT_KNOBS->STORAGE_METRICS_SHARD_LIMIT));
-		Standalone<RangeResultRef> result;
-		for (const auto& ddMetricsRef : resultWithoutPrefix) {
-			// each begin key is the previous end key, thus we only encode the begin key in the result
-			KeyRef beginKey = ddMetricsRef.beginKey.withPrefix(ddStatsRange.begin, result.arena());
-			// Use json string encoded in utf-8 to encode the values, easy for adding more fields in the future
-			json_spirit::mObject statsObj;
-			statsObj["shard_bytes"] = ddMetricsRef.shardBytes;
-			std::string statsString =
-			    json_spirit::write_string(json_spirit::mValue(statsObj), json_spirit::Output_options::raw_utf8);
-			ValueRef bytes(result.arena(), statsString);
-			result.push_back(result.arena(), KeyValueRef(beginKey, bytes));
+	loop {
+		try {
+			auto keys = kr.removePrefix(ddStatsRange.begin);
+			Standalone<VectorRef<DDMetricsRef>> resultWithoutPrefix = wait(
+			    waitDataDistributionMetricsList(ryw->getDatabase(), keys, CLIENT_KNOBS->STORAGE_METRICS_SHARD_LIMIT));
+			Standalone<RangeResultRef> result;
+			for (const auto& ddMetricsRef : resultWithoutPrefix) {
+				// each begin key is the previous end key, thus we only encode the begin key in the result
+				KeyRef beginKey = ddMetricsRef.beginKey.withPrefix(ddStatsRange.begin, result.arena());
+				// Use json string encoded in utf-8 to encode the values, easy for adding more fields in the future
+				json_spirit::mObject statsObj;
+				statsObj["shard_bytes"] = ddMetricsRef.shardBytes;
+				std::string statsString =
+				    json_spirit::write_string(json_spirit::mValue(statsObj), json_spirit::Output_options::raw_utf8);
+				ValueRef bytes(result.arena(), statsString);
+				result.push_back(result.arena(), KeyValueRef(beginKey, bytes));
+			}
+			return result;
+		} catch (Error& e) {
+			if (e.code() == error_code_operation_failed) {
+				TraceEvent(SevWarnAlways, "DataDistributorNotPresent")
+				    .detail("Operation", "DDMetricsReqestThroughSpecialKeys");
+				continue;
```

Comment:
We should probably some kind of backoff strategy here. At least
`wait(delayJittered(FLOW_KNOBS->PREVENT_FAST_SPIN_DELAY));` and maybe even more throttling

Reply:
Thanks for pointing out this.
Done.